literal/decimal_util.go

```diff
@@ -0,0 +1,68 @@
+package literal
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/cockroachdb/apd"
+)
+
+var decimalPattern = regexp.MustCompile(`^[+-]?[\d]{0,38}(\.[\d]{0,38})?$`)
+
+// decimalStringToBytes converts a decimal string to a 16-byte byte array.
+// 16-byte bytes represents a little-endian 128-bit integer, to be divided by 10^Scale to get the decimal value.
+// This function also returns the precision and scale of the decimal value.
+// The precision is the total number of digits in the decimal value. The precision is limited to 38 digits.
+// The scale is the number of digits to the right of the decimal point. The scale is limited to the precision.
+func decimalStringToBytes(decimalStr string) ([16]byte, int32, int32, error) {
+	var result [16]byte
+
+	strings.Trim(decimalStr, " ")
+	if !decimalPattern.MatchString(decimalStr) {
+		return result, 0, 0, fmt.Errorf("invalid decimal string")
+	}
```

Comment:
does the `apd.NewFromString` perform this validation for us such that we don't need to explicitly do it ourselves?

Reply:
not exactly it supports scientific notation, inf, nan